shuup/admin/static_src/product/edit_media.js

```diff
@@ -82,21 +82,40 @@ $(function() {
         $html.insertBefore($source);
     }
 
-    function onDropzoneQueueComplete(dropzone, kind) {
-        if(location.pathname.indexOf("new") > 0) {
-            // save product media the traditional way via the save button when creating a new product
-            return;
-        }
-        const productId = $("#product-" + kind + "-section-dropzone").data().product_id;
+    function getFileIds(kind) {
         const $fileInputs = $("#product-" + kind + "-section").find(".file-control input");
         var fileIds = [];
-
         for(var i = 0; i < $fileInputs.length; i++){
             let fileId = parseInt($($fileInputs[i]).val());
             if(!isNaN(fileId)) {
                 fileIds.push(parseInt($($fileInputs[i]).val()));
             }
         }
+        return fileIds
+    }
+
+    function getFileCount(kind) {
+        return $("#product-" + kind + "-section").data("saved_file_count") || 0;
+    }
+
+    function setFileCount(kind, count) {
+        $("#product-" + kind + "-section").data("saved_file_count", count);
+    }
+
+    function onDropzoneQueueComplete(dropzone, kind) {
+        if(location.pathname.indexOf("new") > 0) {
```

Comment:
I don't think this is a good idea. Can't we have a flag for this coming from the widged instead of relying on the URL?